generator/templates/offline_audio_context.tmpl.rs

```diff
@@ -0,0 +1,301 @@
+
+use std::fs::File;
+use napi::*;
+use napi_derive::js_function;
+use web_audio_api::context::*;
+use crate::*;
+// use web_audio_api::AudioBuffer;
+
+
+pub(crate) struct NapiOfflineAudioContext(OfflineAudioContext);
```

Comment:
I realize that `start_rendering` consuming `self` may be fundamentally incompatible with JS way of working.
You could try to redefine the struct as
`struct NapiOfflineAudioContext(RefCell<Option<OfflineAudioContext>>)`
and when `start_rendering` is called, take the context out and leave `None` in place.
In this case I do not think you need to resort to `unsafe` code, which I presume causes a double drop of the context (hence segfault)

Reply:
Alternatively, we could expose a non-consuming version of the `startRendering` method and hide it behind an undocumented feature flag in the parent library

Reply:
Thank for the ideas! 
I think we should really try the first solution before considering adding special things to the `rs` lib.

I only wonder what should vs what actually happen if we do something like:
```rs
let context = OfflineAudioContext::new();
// later
let buffer = context.start_rendering();
// later
context.decode_audio_data()
```
(agreed that it would be quite weird code)

Reply:
Seems to work with a simple `struct NapiOfflineAudioContext(Option<OfflineAudioContext>)`, thanks!

Reply:
Great!

> I only wonder what should vs what actually happen if we do something like:

I think this is our punishment for deviating from the spec.
We could still roll back the change and make startRendering take `&self`.
In that case, it should return an `Result<AudioBuffer>` because we should throw an `InvalidStateError` when rendering has already started

Reply:
> In that case, it should return an Result<AudioBuffer> because we should throw an InvalidStateError when rendering has already started

Yup, no strong opinion on my side, in one way that would make the API aligned with `decode_audio_audio` which is not really chocking, on the other way it would complicate a bit the user code. In any case, this would be very easy to change here if needed